src/main/java/com/skytask/listener/Listener.java

```diff
@@ -0,0 +1,33 @@
+package com.skytask.listener;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.type.CollectionType;
+import com.skytask.channel.ProductSource;
+import com.skytask.model.Product;
+import com.skytask.service.ProductService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.annotation.StreamListener;
+
+import java.io.IOException;
+import java.util.List;
+
+@EnableBinding(ProductSource.class)
+public class Listener {
+
+    private static Logger logger = LoggerFactory.getLogger(Listener.class);
+    private ProductService productService;
+
+    public Listener(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @StreamListener("responseProductListChannel")
+    public void updateProductList(String products) throws IOException {
+        logger.info("I received: {}", products);
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
Can't you create a public utility class to reuse the same Objecta mapper instance and avoid having to instantiate it every time you want to use it?

Try to use generics.